dms/models/directory.py

```diff
@@ -510,6 +516,26 @@ def _onchange_category(self):
             )
             record.tag_ids = tags
 
+    @api.onchange("model_id")
+    def _onchange_model_id(self):
+        for record in self:
+            if record.model_id:
+                record.res_model = record.model_id.model
+
+    @api.onchange("storage_id")
+    def _onchange_storage_id(self):
+        for record in self:
+            if (
+                not record.is_root_directory
+                and record.storage_id.save_type == "attachment"
+                and record.storage_id.inherit_access_from_parent_record
+            ) or (
+                record.is_root_directory
+                and record.root_storage_id.save_type == "attachment"
+                and record.root_storage_id.inherit_access_from_parent_record
+            ):
```

Comment:
In harmony with other suggestions:

```suggestion
            if record.storage_id.inherit_access_from_parent_record:
```